File: Dataset/Leetcode/train/7/584.cpp
 class Solution {
public:
    int XXX(int x) {
        int temp = x % 10;
        while(x /= 10){
            if(temp > 214748364 || temp < -214748364)
                return 0;
            temp = 10 * temp + x % 10;
        }
        return temp;
    }
};

File: Dataset/Leetcode/valid/102/584.py
class Solution:
    def XXX(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        cur_layer, res = [root], []
        while cur_layer:
            res.append([node.val for node in cur_layer])
            next_layer = []
            for node in cur_layer:
                if node.left:
                    next_layer.append(node.left)
                if node.right:
                    next_layer.append(node.right)
            cur_layer = next_layer
        return res

